packages/wallet/src/redux/protocols/advance-channel/reducer.ts

```diff
@@ -201,6 +234,33 @@ const channelUnknownReducer: ProtocolReducer<states.AdvanceChannelState> = (
   } else {
     return { protocolState, sharedData };
   }
+}
+
+const channelUnknownReducer: ProtocolReducer<states.AdvanceChannelState> = (
+  protocolState: states.ChannelUnknown,
+  sharedData,
+  action: CommitmentsReceived,
+) => {
+  const { privateKey } = protocolState;
+  const channelId = getChannelId(action.signedCommitments[0].commitment);
+  const checkResult = checkAndInitialize(sharedData, action.signedCommitments[0], privateKey);
+  if (!checkResult.isSuccess) {
+    throw new Error('Could not initialize channel');
+  }
+  sharedData = checkResult.store;
+  sharedData = checkCommitments(sharedData, 0, action.signedCommitments);
+
+  const result = attemptToAdvanceChannel(sharedData, protocolState, channelId);
```

Comment:
Should we use destructuring assignment here?

Reply:
Destructuring is pretty awkward here:
- You can't overwrite `protocolState` because it might change the type
- You can't re-declare `sharedData` as a const
- If you declare a new `nextSharedData` variable, you have to then assign `sharedData = nextSharedData`, or do something else, to take into account that you might then register the channel to monitor on L260